src/ui/Histogram/Histogram.js

```diff
@@ -0,0 +1,692 @@
+    ctx.lineTo(xMax, yMax);
+    ctx.stroke();
+
+    ctx.font = '10px Arial';
+    ctx.fillStyle = 'rgb(120, 20, 20)';
+    ctx.textBaseline = 'top';
+    ctx.textAlign = 'center';
+
+    let maxHistValue = 1.0;
+    if (m_peakIndex > 0) {
+      maxHistValue = m_histogram[m_peakIndex] * 2;
+      maxHistValue = maxHistValue > 1.0 ? 1.0 : maxHistValue;
+    }
+    let i;
+    const NUM_X_MARKS = 4;
+    for (i = 0; i <= NUM_X_MARKS; i++) {
+      const x = xMin + Math.floor((wRect * i) / NUM_X_MARKS);
+      ctx.moveTo(x, yMax);
+      ctx.lineTo(x, yMax + 6);
+      ctx.stroke();
+      const valMark = Math.floor(0 + (m_numColors * i) / NUM_X_MARKS);
+      if (i === 0) {
+        ctx.textAlign = 'left';
+      } else if (i === NUM_X_MARKS) {
+        ctx.textAlign = 'right';
+      } else {
+        ctx.textAlign = 'center';
+      }
+      ctx.fillText(valMark.toString(), x, yMax + 4);
+    }
+
+    if (NEED_TO_DRAW_VERTICAL_MARKS) {
+      ctx.textBaseline = 'bottom';
+      ctx.textAlign = 'left';
+      ctx.fillStyle = 'rgb(120, 60, 60)';
+      const NUM_Y_MARKS = 4;
+      for (i = 0; i <= NUM_Y_MARKS; i++) {
+        if (i === NUM_Y_MARKS) {
+          ctx.textBaseline = 'top';
+        }
+        const y = yMax - Math.floor((hRect * i) / NUM_Y_MARKS);
+        ctx.moveTo(xMin, y);
+        ctx.lineTo(xMin - 4, y);
+        ctx.stroke();
+        const valMark = 0 + (maxHistValue * i) / NUM_Y_MARKS;
+        ctx.fillText(valMark.toFixed(2), xMin + 6, y);
+      }
+    }
+
+    ctx.lineWidth = 2;
+    ctx.strokeStyle = '#080808';
+    ctx.fillStyle = '#707070';
+
+    ctx.beginPath();
+    {
+      ctx.moveTo(xMin, yMax);
+      let i;
+      let x, y;
+      for (i = 0; i < m_numColors; i++) {
+        x = xMin + Math.floor((wRect * i) / m_numColors);
+        let v = m_histogram[i] / maxHistValue;
+        v = v >= 1.0 ? 1.0 : v;
+        y = yMax - Math.floor(hRect * v);
+        ctx.lineTo(x, y);
+      }
+      y = yMax;
+      ctx.lineTo(x, y);
+    }
+    ctx.closePath();
+    ctx.fill();
+    // draw peak
+    if (m_peakIndex > 0) {
+      ctx.lineWidth = 1;
+      ctx.strokeStyle = '#eeeeee';
+      const x = xMin + Math.floor((wRect * m_peakIndex) / m_numColors);
+      let v = m_histogram[m_peakIndex] / maxHistValue;
+      v = v >= 1.0 ? 1.0 : v;
+      let y = yMax - Math.floor(hRect * v);
+      ctx.beginPath();
+      ctx.setLineDash([5, 15]);
+      ctx.moveTo(x, y);
+      y = yMax;
+      ctx.lineTo(x, y);
+      ctx.stroke();
+      ctx.setLineDash([]);
+    }
+    // render points and lines for modified transfer fucntion
+    if (m_transfFuncCallback !== undefined && m_transfFuncUpdateCallback !== undefined) {
+      m_transfFunc.render(ctx, xMin, yMin, wRect, hRect);
+    }
+  }
+
+  const vol = props.volume;
+  if (vol === undefined) {
+    return <p>UiHistogram.props volume is not defined !!!</p>;
+  }
+  if (vol === null) {
+    return <p></p>;
+  }
+
+  m_transfFuncCallback = props.transfFunc;
```

Comment:
It will be better to use transFunc and transFuncUpdate from props without creating variable